google/cloud/pubsub/publisher_connection_test.cc

```diff
@@ -364,9 +364,13 @@ TEST(MakePublisherConnectionTest, TracingEnabled) {
       publisher->Publish({MessageBuilder{}.SetData("test-data-0").Build()})
           .get();
 
+  auto spans = span_catcher->GetSpans();
   EXPECT_THAT(
-      span_catcher->GetSpans(),
+      spans,
       Contains(SpanNamed("projects/test-project/topics/test-topic send")));
```

Comment:
Can this be one expectation?

Ideally we could use `::testing::ElementsAre`. But if not, we could use `::testing::IsSupersetOf` or `::testing::AllOf(Contains(foo), Contains(bar))`

Reply:
Yeah I initially tried using ElementsAre
```  
EXPECT_THAT(
      span_catcher->GetSpans(),
      ::testing::ElementsAre(SpanNamed("projects/test-project/topics/test-topic send"),SpanNamed(
                  "pubsub::FlowControlledPublisherConnection::Publish")));
```

Produces a failure:
```
/usr/local/google/home/alevenb/google/google-cloud-cpp/google/cloud/pubsub/publisher_connection_test.cc:370: Failure
Value of: span_catcher->GetSpans()
Expected: has 2 elements where
element #0 has name: projects/test-project/topics/test-topic send,
element #1 has name: pubsub::FlowControlledPublisherConnection::Publish
  Actual: { (ptr = 0x557d1f2802b0, value = Span {name=pubsub::FlowControlledPublisherConnection::Publish, kind=CLIENT, instrumentation_scope {gcloud-cpp, v2.16.0-rc}, attributes=[gcloud.status_code=std::int32_t:0], events=[]}), (ptr = 0x557d1f284700, value = Span {name=projects/test-project/topics/test-topic send, kind=PRODUCER, instrumentation_scope {gcloud-cpp, v2.16.0-rc}, attributes=[messaging.message_id=std::string:test-message-id-0, messaging.message.total_size_bytes=std::int64_t:31, messaging.pubsub.ordering_key=std::string:, gcloud.status_code=std::int32_t:0, messaging.destination.template=std::string:topic, messaging.destination.name=std::string:projects/test-project/topics/test-topic, messaging.system=std::string:pubsub], events=[]}) }, whose element #0 doesn't match, has name: pubsub::FlowControlledPublisherConnection::Publish
```

Both isSuperSetOf or AllOf work.

```
  EXPECT_THAT(
      span_catcher->GetSpans(),
      ::testing::IsSupersetOf({
          SpanNamed("projects/test-project/topics/test-topic send"),
          SpanNamed("pubsub::FlowControlledPublisherConnection::Publish")}));
  EXPECT_THAT(
      span_catcher->GetSpans(),
      ::testing::AllOf(
          Contains(SpanNamed("projects/test-project/topics/test-topic send")),
          Contains(SpanNamed("pubsub::FlowControlledPublisherConnection::Publish"))));
```

Looks like UnorderedElementsAre works too... using that since I think it's the clearest.